src/nunit.analyzers.tests/ConstraintsUsage/StringConstraintUsageCodeFixTests.cs

```diff
@@ -121,5 +130,58 @@ public void AnalyzeStringBooleanMethodAssertThatIsFalse(string method, string an
 
             RoslynAssert.CodeFix(analyzer, fix, ExpectedDiagnostic.Create(analyzerId), code, fixedCode);
         }
+
+        [TestCaseSource(nameof(NegativeAssertData))]
+        public void CodeFixMaintainsReasonableTriviaWithEndOfLineClosingParen(string method, string analyzerId, string suggestedConstraint)
+        {
+            var code = TestUtility.WrapInTestMethod($@"
+            ClassicAssert.IsFalse(
+                ↓""abc"".{method}(""ab""));");
+
+            var fixedCode = TestUtility.WrapInTestMethod($@"
+            Assert.That(
+                ""abc"",
+                {suggestedConstraint}(""ab""));");
+
+            RoslynAssert.CodeFix(analyzer, fix, ExpectedDiagnostic.Create(analyzerId), code, fixedCode);
+        }
+
+        [TestCaseSource(nameof(NegativeAssertData))]
+        public void CodeFixMaintainsReasonableTriviaWithNewLineClosingParen(string method, string analyzerId, string suggestedConstraint)
+        {
+            var code = TestUtility.WrapInTestMethod($@"
+            Assert.That(
+                ↓""abc"".{method}(""ab""),
+                Is.False
+            );");
+
+            var fixedCode = TestUtility.WrapInTestMethod($@"
+            Assert.That(
+                ""abc"",
+                {suggestedConstraint}(""ab"")
+            );");
+
+            RoslynAssert.CodeFix(analyzer, fix, ExpectedDiagnostic.Create(analyzerId), code, fixedCode);
+        }
+
+        [Test]
+        public void CodeFixMaintainsReasonableTriviaWithAllArgumentsOnSameLine(
+            [ValueSource(nameof(StringConstraints))] string method,
+            [Values] bool newlineBeforeClosingParen)
+        {
+            var analyzerId = NegativeAssertDictionary[method][1];
+            var suggestedConstraint = NegativeAssertDictionary[method][2];
+            var optionalNewline = newlineBeforeClosingParen ? @"
+            " : string.Empty;
```

Comment:
Doesn't have to use a literal string here.

```suggestion
            var optionalNewline = newlineBeforeClosingParen ? "\n" : string.Empty;
```